apps/analytics/views.py

```diff
@@ -205,3 +207,48 @@ def get_last_submission_datetime_analysis(request, challenge_pk, challenge_phase
     except:
         response_data = {'error': 'Bad request. Please try again later!'}
         return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['GET', ])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_challenge_phase_submission_analysis(request, challenge_pk, challenge_phase_pk):
+    """
+    Returns
+    1. Total number of submissions in a challenge phase
+    2. Number of teams which made submissions in a challenge phase
+    3. Number of submissions with status a)Submitting, b)Submitted, c)Running, d)Failed, e)Cancelled, f)Finished status
+    4. Number of flagged & public submissions in challenge phase
+    """
+
+    challenge = get_challenge_model(challenge_pk)
```

Comment:
nit: Please remove all the newlines between L225 and L241.

Reply:
@deshraj Is there any specific reason for it? I added them in order to make the code comfortable while reading.

Reply:
The logic is pretty straightforward and is easy to understand so doesn't need explicit new lines. Moreover, we have comments.

Reply:
Looks much better now. Thanks!